### PR TITLE
Reduce .rubocop_todo.yml

### PR DESCRIPTION
**Pull request by walf443**
This PR reduce .rubocop_todo.yml

 * RSpec/ScatteredSetup
 * RSpec/ReturnFromStub
 * RSpec/RepeatedExample
 * Rspec/DescribeClass
 * RSpec/BeforeAfterAll
 * RSpec/ExpectInHook
 * RSpec/IteratedExpectation

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [ ] Squashed related commits together.
* [x] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests(`rake spec`) are passing.
* [x] The new code doesn't generate RuboCop offenses that are checked by `rake internal_investigation`.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/style/trailing_underscore_variable_spec.rb [hunk 1]
```diff
@@ -224,11 +224,6 @@
 
     include_examples 'common functionality'
 
-    it 'does not register an offense for an underscore variable preceded ' \
-       'by a named splat underscore variable' do
-      expect_no_offenses('a, *_b, _ = foo()')
```
Comment (walf443) [inline spec/rubocop/cop/style/trailing_underscore_variable_spec.rb]: This spec duplicated of https://github.com/bbatsov/rubocop/pull/4989/files#diff-7fe7950ba35a008e696293c0f39bf18bL244

File: spec/rubocop/cli/cli_options_spec.rb [hunk 2]
```diff
@@ -1006,67 +1006,72 @@ def f
     end
 
     after do
```
Comment (Drenmi) [inline spec/rubocop/cli/cli_options_spec.rb]: We're left with an empty block here. Probably want to remove that?

File: spec/rubocop/cli/cli_options_spec.rb [hunk 3]
```diff
@@ -1006,67 +1006,72 @@ def f
...
+    end
```
Comment (pocke) [inline spec/rubocop/cli/cli_options_spec.rb]: Remove this `after` 😉

File: spec/project_spec.rb [hunk 4]
```diff
@@ -146,9 +142,7 @@
             entry.match(/^\*\s*\[/)
           end
 
-          entries_including_issue_link.each do |entry|
-            expect(entry).to include('): ')
-          end
+          expect(entries_including_issue_link).to all(be_include('): '))
```
Comment (bquorning) [inline spec/project_spec.rb]: `be_include` doesn’t sound right. But I’m not sure what would be another way to fix it.

File: spec/rubocop/cli/cli_options_spec.rb (hunk 3 above)
Comment (bbatsov) [reply spec/rubocop/cli/cli_options_spec.rb]: @backus I think you need to create a cop for that. ;-) Or we should do it. :D

Comment (bbatsov) [reply spec/rubocop/cli/cli_options_spec.rb]: Probably it should be us.

File: spec/project_spec.rb (hunk 4 above)
Comment (mikegee) [reply spec/project_spec.rb]: `all(include('): '))` might work. I'm pretty sure I've used `all(include(...))` before.

**Comment by bbatsov** [comment]
Thanks!

**Comment by backus** [comment]
Love this PR. Nice job!

**Pull request merged**